### DISCUSSION TITLE
[suggestion]: make Angledraw sctrl capable of scaling hitboxes

### DISCUSSION THREAD

**Discussion by SaltAddict** [ikemen-engine/Ikemen-GO#919] [Feature Request] [answered]
Angledraw controller has a scale parameter but this doesn't change hitbox size the way setting the scale in the constants does. Would be cool if there was a new parameter to enable realtime hitbox scaling as it would make doing beams and other such things a lot easier.

**Comment by Sluigi123** [comment]
This is a nice idea. However, I can tell that it'll be a bit tedious though. Regardless, this is definitely doable.
Besides, this suggestion was made a bit ago; yet, having more people making the same suggestion creates more motivation and confidence for the developers.

**Comment by Isakku** [comment]
The idea sounds awesome, But it would have to be a newer parameter in the AngleDraw Stctrl.
Actually, AngleDraw takes two parameters to scale a sprite, if you want to scale CLNS boxes, two additional X and Y scale parameters would be needed. I must say this reminds me of a suggestion I have for scaling:
https://github.com/ikemen-engine/Ikemen-GO/issues/639

**Comment by potsmugen** [comment, marked as answer]
Implemented in `TransformClsn`.
